### PR TITLE
Fix theming of material preview icons

### PR DESCRIPTION
**Pull request by Chaosus**
Fixes to properly updates these icons at theme changing:

image

### CODE REVIEW

File: editor/plugins/material_editor_plugin.cpp [hunk 1]
```diff
@@ -39,24 +39,16 @@
...
-				light_2_switch->set_pressed_texture(get_theme_icon(SNAME("MaterialPreviewLight2Off"), SNAME("EditorIcons")));
-
-				sphere_switch->set_normal_texture(get_theme_icon(SNAME("MaterialPreviewSphereOff"), SNAME("EditorIcons")));
-				sphere_switch->set_pressed_texture(get_theme_icon(SNAME("MaterialPreviewSphere"), SNAME("EditorIcons")));
-				box_switch->set_normal_texture(get_theme_icon(SNAME("MaterialPreviewCubeOff"), SNAME("EditorIcons")));
-				box_switch->set_pressed_texture(get_theme_icon(SNAME("MaterialPreviewCube"), SNAME("EditorIcons")));
-
-				first_enter = false;
-			}
+		case NOTIFICATION_THEME_CHANGED: {
```
Comment (akien-mga) [inline editor/plugins/material_editor_plugin.cpp:42]: Does `THEME_CHANGED` get called also when instantiating a new editor plugin? I think usually we make such logic run both for `READY` and `THEME_CHANGED` to handle both, no?

Comment (Chaosus) [reply editor/plugins/material_editor_plugin.cpp:42]: ah, I didn't remember really about this part of the engine - I will return a NOTIFICATION_ENTER_TREE instead. But when you simply runned an editor its calling THEME_CHANGED which is strange a bit.

Comment (AaronRecord) [reply editor/plugins/material_editor_plugin.cpp:43]: @akien-mga I don't think it's necessary, see https://github.com/godotengine/godot/issues/50743. I made an experimental [commit](https://github.com/AaronRecord/godot/commit/960fd97f8a42b9f62ef501717095ef6d81286e1b) where I removed notification enter tree, and one of the test cases fails, but the editor starts and closes and seems to run fine. I'm hoping to clean up the theme propagation logic but I'm currently away from home for a few days so I can't work on it.

**Comment by fire** [comment]
Would like to see this bug fixed. Will see what I can do to review.

**Pull request closed without merging**